synapseclient/core/download/download_functions.py

```diff
@@ -161,6 +170,105 @@ async def download_file_entity(
     entity.cacheDir = os.path.dirname(download_path)
 
 
+async def download_file_entity_model(
+    *,
+    download_location: str,
+    file: "File",
+    if_collision: str,
+    submission: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> None:
+    """
+    Download file entity
+
+    Arguments:
+        download_location: The download location
+        entity:           The File object
+        if_collision:      Determines how to handle file collisions.
+                            May be
+
+            - `overwrite.local`
+            - `keep.local`
+            - `keep.both`
+
+        submission:       Access associated files through a submission rather than through an entity.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    # set the initial local state
+    file.path = None
+
+    # check to see if an UNMODIFIED version of the file (since it was last downloaded) already exists
+    # this location could be either in .synapseCache or a user specified location to which the user previously
+    # downloaded the file
+    cached_file_path = client.cache.get(
+        file_handle_id=file.data_file_handle_id, path=download_location
```

Comment:
Let's say I download a file `myfile.csv` and then rename it something later called `myfile_original.csv` before downloading the entity again (the entity has not changed). Because this is searching for the file via the `file_handle_id` and not the name of the file on disk, this would not lead to a download that overwrites the `myfile_original.csv` correct?

Reply:
So it's finding the path by `file_handle_id` and also the `download_location` passed in. In this case it would be a cache miss and it would download the file again. Example:
![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/7a825622-e839-4805-8030-a85484973565)

Reply:
Awesome, thanks for testing this :)